rotkehlchen/db/upgrades/v26_v27.py

```diff
@@ -35,6 +33,7 @@ def upgrade_v26_to_v27(db: 'DBHandler') -> None:
     PRIMARY KEY (tx_hash, log_index)
 );
 """)
+    cursor.execute('DROP TABLE IF EXISTS balancer_pools;')
```

Comment:
So @yabirgb moving the deletion of balancer events, before the pools so that the constraint does not hit was the actual bug fix.

Also added test data and modified the existing test to test for this.